### PR TITLE
Developer-specific flag -1 was added for Loop_library::set().

### PR DESCRIPTION
**Pull request by uukhas**
This is an improvement of the loop library interface, connected to #198.

### CODE REVIEW

**Comment by Expander** [comment]
That looks perfect!

Is it possible to add a very simple test (for a very simple model like the `SM`), which checks that the spectrum generator is run with the library specified by `FLEXIBLESUSY_LOOP_LIBRARY`?

Could you also please add a short documentation of this new environment variable?

**Comment by uukhas** [comment]
I think that it is even possible to add a test without a connection to any model.

**Comment by Expander** [comment]
> I think that it is even possible to add a test without a connection to any model.

That would be much better!

File: src/loop_libraries/loop_library.hpp [hunk 1]
```diff
@@ -30,6 +30,7 @@ class Loop_library
 public:
    enum class Library { Undefined, Softsusy, Collier, Looptools, Fflite };
    static void set(int);
+   static Library get_type();
```
Comment (Expander) [inline src/loop_libraries/loop_library.hpp:33]: Please make this function `const`.

File: test/test_looplibrary_environment.sh [hunk 2]
```diff
@@ -0,0 +1,38 @@
+#!/bin/sh
+
+exit_code=0
+
+exe_test () {
+   TEST_LOOPLIBRARY=$(./test_looplibrary_environment.x \
```
Comment (Expander) [inline test/test_looplibrary_environment.sh:6]: I think this executable is currently missing.

Comment (uukhas) [reply test/test_looplibrary_environment.sh:6]: 0b867f0

File: src/loop_libraries/loop_library.hpp (hunk 1 above)
Comment (uukhas) [reply src/loop_libraries/loop_library.hpp]: a8903e3

File: src/loop_libraries/loop_library.cpp [hunk 3]
```diff
@@ -109,7 +109,7 @@ void Loop_library::set(int new_type)
    }
 }
 
-Loop_library::Library Loop_library::get_type()
+const Loop_library::Library Loop_library::get_type()
```
Comment (Expander) [inline src/loop_libraries/loop_library.cpp:112]: Sorry, I my proposal was probably confusing. :)
Please make the *function* (i.e. the `this` pointer) `const`, not the return type, such that the function can be called on `const` objects etc.:
~~~
Loop_library::Library Loop_library::get_type() const
~~~

Comment (uukhas) [reply src/loop_libraries/loop_library.cpp:112]: @Expander, this function is a _static_ one. It seems, that static const functions are not allowed in C++. This is why I thougth, that you meant to make the _return_ type const.

Comment (Expander) [reply src/loop_libraries/loop_library.cpp:112]: Ouch! Sorry for my stupidity, I missed that. :) (The `type` it returns confused me, but it is also `static`.) Please ignore my comment and revert commit a8903e3 .

**Comment by uukhas** [comment]
@Expander, I also have a question: why generic rule for all `.x` files
```
$(DIR)/test_%.x: $(DIR)/test_%.o
```
is ignored in `all-test`, if one adds
```
$(DIR)/test_looplibrary_environment.sh.xml : $(DIR)/test_looplibrary_environment.x

```

**Comment by uukhas** [comment]
@Expander, Is there a possibility to get somehow more information from `travis`? 

For example, now the new test fails on `macOS`, but I cannot debug it, because I do not have any useful information except the fact that tests failed.

**Comment by Expander** [comment]
You can (temporarily, i.e. in this PR) append the following line to `.travis.yml` to execute a specific test:
~~~
  - test/test_<...>.sh [flags]
~~~
After the problem has been fixed, one can simply remove the line again.

**Comment by Expander** [comment]
Sorry, the typo was my mistake!

**Comment by uukhas** [comment]
> Sorry, the typo was my mistake!

@Expander, no it was mine: I am making debugs on `travis` with my test only... and I managed to write a name for the test in a wrong way.

**Comment by uukhas** [comment]
Actually, it seems that the problem is inside `grep`s flag `-P`. I think that it is not supported by `macOS`.

**Comment by uukhas** [comment]
It seems that everything is OK now.

**Comment by Expander** [comment]
Very nice, thanks a lot!
@wkotlarski are you OK with this change?

File: src/loop_libraries/loop_library.cpp [hunk 4]
```diff
@@ -59,11 +60,19 @@ void Loop_library::set_default()
...
    Loop_library::type_ = Loop_library::Library::Softsusy;
+   VERBOSE_MSG("Enabling softsusy.");
 }
 
 void Loop_library::set(int new_type)
 {
    if (Loop_library::type_ == Loop_library::Library::Undefined) {
+      if (new_type == -1) {
+         if (const char* flag = std::getenv("FLEXIBLESUSY_LOOP_LIBRARY")) {
+            new_type = std::atoi(flag);
```
Comment (wkotlarski) [inline src/loop_libraries/loop_library.cpp:71]: Maybe we could wrap above line in a try-catch blocks? And print something meaningful if the variable is not set/contains not a numer.

Comment (uukhas) [reply src/loop_libraries/loop_library.cpp:71]: If it is not a number it returns 0, i.e. `softsusy`

Comment (uukhas) [reply src/loop_libraries/loop_library.cpp:71]: And exactly one line below it is printed, in case of `--enable-verbose`.

Comment (Expander) [reply src/loop_libraries/loop_library.cpp:71]: Ah, ok, so this nicely fits together. :)

**Pull request merged**